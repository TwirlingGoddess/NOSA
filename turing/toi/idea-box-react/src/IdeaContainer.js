import React from 'react';
import IdeaCard from './IdeaCard';

const IdeaContainer = ({ideas, removeIdea}) => {
	const details = ideas.map(idea => {
		<IdeaCard 	{...idea}
					key={idea.id}
					removeIdea={removeIdea}
		/>
	})

	return(
		<div className='Container'>
			{details}
		</div>
	)
}

export default IdeaContainer