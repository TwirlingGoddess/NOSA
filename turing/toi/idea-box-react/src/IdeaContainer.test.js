import React from 'react';
import { shallow } from 'enzyme';
import IdeaContainer from './IdeaContainer';

describe('IdeaContainer', () => {
	let mockIdeas
	let wrapper

	it('should render all idea cards', () => {
		mockIdeas = [{ title: 'Testing1', body: 'I used to hate it, but now I love it.', id: Date.now() },
					{ title: 'Testing2', body: 'Now that I love it, I am unstoppable.', id: Date.now() }]
		wrapper = shallow(<IdeaContainer ideas={mockIdeas} />)
		expect(wrapper.find('Container')).children.length.toBe(2)
	})

	it('should match the snapshot', () => {
		mockIdeas = [{ title: 'Testing1', body: 'I used to hate it, but now I love it.', id: Date.now() },
					{ title: 'Testing2', body: 'Now that I love it, I am unstoppable.', id: Date.now() }]
		wrapper = shallow(<IdeaContainer ideas={mockIdeas} />)		
		expect(wrapper).toMatchSnapshot()
	})
})