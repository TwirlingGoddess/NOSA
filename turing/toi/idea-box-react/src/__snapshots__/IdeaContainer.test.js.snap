// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaContainer
    ideas={
        Array [
            Object {
              "body": "I used to hate it, but now I love it.",
              "id": 1534387277050,
              "title": "Testing1",
            },
            Object {
              "body": "Now that I love it, I am unstoppable.",
              "id": 1534387277050,
              "title": "Testing2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
